src/main/java/frc/robot/RobotContainer.java

```diff
@@ -40,25 +43,30 @@ public RobotContainer() {
         // The left stick controls translation of the robot.
         // Turning is controlled by the X axis of the right stick.
         new RunCommand(
-            () ->
-                m_robotDrive.drive(
-                    new Vector(
-                        MathUtil.applyDeadband(
-                            -m_driverController.getLeftY(), OIConstants.kDriveDeadband),
-                        MathUtil.applyDeadband(
-                            -m_driverController.getLeftX(), OIConstants.kDriveDeadband)),
-                    new Vector(
-                        MathUtil.applyDeadband(
-                            -m_driverController.getRightX(), OIConstants.kDriveDeadband),
-                        MathUtil.applyDeadband(
-                            -m_driverController.getRightY(), OIConstants.kDriveDeadband)),
-                    m_driverController.getRightBumper(),
-                    m_driverController.getAButton()),
+            () -> {
+              updateInput();
+              m_robotDrive.drive(
+                  leftInputVec,
+                  rightInputVec,
+                  m_driverController.getRightBumper(),
+                  m_driverController.getAButton());
+            },
             m_robotDrive));
 
     SmartDashboard.putData("Auto Chooser", autoChooser);
   }
 
+  private void updateInput() {
+    leftInputVec.setX(
+        MathUtil.applyDeadband(-m_driverController.getLeftY(), OIConstants.kDriveDeadband));
```

Comment:
Why is your X set to Y?

Reply:
That’s a big, I’ll fix it

Reply:
Are you sure that's a bug?

Reply:
I just checked the old code and I guess it's not, though it might be cleaner to rotate the vector rather than swapping the x and y inputs

Reply:
I agree that it can (and should) be more clear that this is a transformation from "Joystick XY" to "Robot XY", which is really a projection into the standard 3D coordinate system.

Reply:
@ChainedGCH this is because we're transforming "Joystick XY" into "Robot XY(Z)". Robots operate in 3-D space and so use that coordinate system.

Reply:
@ChainedGCH Yes their "x" is a different X. It's the same X as in the right hand rule. There are many different choices one can make, but so long as it's a rotated version of your "right-hand rule" fingers, it's positive orientation. Other orientations are mirrored versions of your right hand rule and are "left-handed" and thus negative orientation. Long story short, the robot coordinate system has X as back-forward, Y as left-right, and Z as up-down. This is a common coordinate system. 

@awfulbananas I think this code should've had a comment mentioning this to avoid this confusion. There's nothing wrong with not rotating explicitly as that's more computationally expensive.